src/TRestRawSignal.cxx

```diff
@@ -603,21 +608,30 @@ void TRestRawSignal::GetWhiteNoiseSignal(TRestRawSignal* noiseSgnl, Double_t noi
 ///
 /// \param averagingPoints It defines the number of neightbour consecutive
 /// points used to average the signal
+/// \param option If option is set to "RAW", the returned signal is the smoothed raw data without baseline correction
 ///
-void TRestRawSignal::GetSignalSmoothed(TRestRawSignal* smthSignal, Int_t averagingPoints) {
```

Comment:
What is the fundamental difference between using "RAW" option or not to use it?

Probably it is best to replace `GetData` by `GetRawData` and thats it?

Reply:
In fact I don't think it makes much sense to use `TRestRawSignal* smthSignal` to store a smoothed signal. Note that TRestRawSignal stores data as `Short_t` while you can get more precision after smoothing `Float_t` or so. I think would be better to return a `std::vector<Float_t>` or so. Another issue is that we should distinguish between processed and raw data.

Reply:
Then, the best is to implement both methods:

```
void TRestRawSignal::GetSignalSmoothed( TRestRawSignal *smoothed, Int_t averagingPoints );
std::vector <Float_t> TRestRawSignal::GetSignalSmoothed( Int_t averagingPoints );
```

The first method keeps the smoothed signal in the raw domain, while the second gives a more accurate version.

Reply:
>>> Another issue is that we should distinguish between processed and raw data.

As soon as we are processing data with the raw library, the data type will be `Short_t`. The moment to get better precision is the moment we transform `TRestRawSignalEvent` into `TRestDetectorSignalEvent`.

Reply:
We will be always be able to identify if the data was processed and how it was processed looking to the processes metadata registered by the processing chain.

Reply:
I removed the "RAW" option from `GetSignalSmoothed` and `GetIntegralInRange` and replaced in both `GetData` with `GetRawData`, and added the overloaded method `std::vector<Float_t> TRestRawSignal::GetSignalSmoothed( Int_t averagingPoints );`

Reply:
It is still quite good at the boundaries, and after all we should allow some margin for our signals to be inside a reduced window with some margins.

It would be great to add a `TRestRawBaselineCorrectionProcess` that corrects the baseline. You could use that picture and the corrected version to illustrate the process.

Reply:
Hi Konrad,
I have a function to perform smoothing that you can largely use, find it below:
```
std::vector<Float_t>  TRestRawSignal::GetSignalSmoothed(const int &neighbours){
std::vector<Float_t> smoothed(GetNumberOfPoints(),0);
    for (int i = 0; i < GetNumberOfPoints(); i++){
    int nPoints=0;
      for (int j = - neighbours; j<=neighbours;j++){
        int index = i + j;
        if(index<0 || index>= GetNumberOfPoints())continue;
        smoothed[i] +=  GetRawData(index);
        nPoints++;
      }
    if(nPoints>0)smoothed[i] /= nPoints;
    }
  return smoothed;
}
```